### PR TITLE
Ion hash

### PR DESCRIPTION
**Pull request by DavidBM**
Finally, the hashing is finished. The test are enough for me to feel somewhat confident that it will work well. Next is testing with production data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+Ion binary is a library written in safe rust for parsing, encoding and hashing Amazon's Ion binary format.
 
 [![Coverage Status](https://coveralls.io/repos/github/Couragium/ion-binary-rs/badge.svg?branch=master)](https://coveralls.io/github/Couragium/ion-binary-rs?branch=master)
 [![Buils Status](https://github.com/Couragium/ion-binary-rs/workflows/Rust/badge.svg)](https://github.com/Couragium/ion-binary-rs/actions)
 [![Documentation](https://docs.rs/ion-binary-rs/badge.svg)](https://docs.rs/ion-binary-rs)
 [![Crates.io](https://img.shields.io/crates/v/ion-binary-rs)](https://crates.io/crates/ion-binary-rs)
 
-It should be able to parse and encode anything you throw at it. Any failure to do so
+It should be able to **parse**, **encode** and **hash** anything you throw at it. Any failure to do so
 is a bug 💥 that we will fix and we will be very happy if you report them 🙌.
```
Comment (voro2038) [inline README.md:13]: shorter sentences, less emojis

File: src/binary_encoder.rs [hunk 2]
```diff
@@ -67,11 +67,38 @@ pub fn encode_datetime(value: &DateTime<FixedOffset>) -> Vec<u8> {
...
+    // Accounting for the case of a leap second, which shouldn't ever happen.
+    // https://docs.rs/chrono/0.4.19/chrono/naive/struct.NaiveTime.html#leap-second-handling
     if nanosecond > 1_000_000_000 {
         nanosecond -= 1_000_000_000;
     }
 
-    let offset = value.offset().utc_minus_local() / 60;
+    // WARNING:
+    //
+    // This is a bit tricky. With this line we ensure that we have 9 decimal position
```
Comment (voro2038) [inline src/binary_encoder.rs:78]: this text needs reviewing

File: src/tests/js_hash/ion_hash_complex.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import * as ion from 'ion-js';
+import * as ionHash from 'ion-hash-js';
+import * as crypto from "crypto";
+
+// This function will print a strict ready to copy past into rust
```
Comment (voro2038) [inline src/tests/js_hash/ion_hash_complex.ts:5]: a wat?

Comment (DavidBM) [reply src/tests/js_hash/ion_hash_complex.ts:5]: This is the code I used in order to generate the computed hash for a test. I left it in here for reproducibility.

**Comment by DavidBM** [comment]
@voro2038 ready to check

**Review by aegis22 — APPROVED** [review]

**Review by voro2038 — APPROVED** [review]

**Pull request merged**